### DISCUSSION TITLE
SOCTOPUS Missing after upgrade to 2.3.70

### DISCUSSION THREAD

**Discussion by 8inaryMata1eao** [Security-Onion-Solutions/securityonion#5524] [Unsupported Versions] [answered]
A few weeks ago I upgraded the grid to 2.3.70.  After that I saw the dashboard showing a fault for the Manager node.  I rebooted everything, ran "docker restart $(sudo docker ps -a -q)" and "salt-call state.highstate" a few times only to see errors that soctopus could not be found.  What could have happened and how do i resolve this? 
Thanks

"SO-STATUS"
![image](https://user-images.githubusercontent.com/80425093/133479543-19b70606-69c9-42a7-b399-a83e699f4d35.png)

"salt-call state.highstate"
![image](https://user-images.githubusercontent.com/80425093/133480252-64f081b1-8663-470e-b4ff-f8860b4b434b.png)

**Comment by bretthanus** [comment]
> 
> 
> A few weeks ago I upgraded the grid to 2.3.70. After that I saw the dashboard showing a fault for the Manager node. I rebooted everything, ran "docker restart $(sudo docker ps -a -q)" and "salt-call state.highstate" a few times only to see errors that soctopus could not be found. What could have happened and how do i resolve this?
> Thanks
> 
> "SO-STATUS"
> ![image](https://user-images.githubusercontent.com/80425093/133479543-19b70606-69c9-42a7-b399-a83e699f4d35.png)
> 
> "salt-call state.highstate"
> ![image](https://user-images.githubusercontent.com/80425093/133480252-64f081b1-8663-470e-b4ff-f8860b4b434b.png)

I am in a very similar place -- also after update to 2.3.70-- except I have playbook and soctopus as missing, and aptcacherng as pending (but never starts)
`
 sudo so-status

Checking Docker status

    Docker ----------------------------------------------------------------------------------------------------------------------------------------- [ OK ]

Checking container statuses

    so-aptcacherng ---------------------------------------------------------------------------------------------------------------------------- [ PENDING ]
    so-cortex -------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-dockerregistry ------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-elastalert ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-elasticsearch ------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-filebeat ------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-fleet --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-grafana ------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-idstools ------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-influxdb ------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-kibana -------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-kratos -------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-logstash ------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-mysql --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-nginx --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-playbook ------------------------------------------------------------------------------------------------------------------------------- [ MISSING ]
    so-redis --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-sensoroni ----------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-soc ----------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-soctopus ------------------------------------------------------------------------------------------------------------------------------- [ MISSING ]
    so-telegraf ------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-thehive ------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-thehive-es ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-wazuh --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
`

if i run so-start playbook or soctopus, the start is successful and reports a state change from 'none' to 'running', 
----------
          ID: so-playbook
    Function: docker_container.running
      Result: True
     Comment: Created container 'so-playbook'
     Started: 14:27:37.564613
    Duration: 1878.496 ms
     Changes:
              ----------
              container_id:
                  ----------
                  added:
                      671e8511e976501833b32c7f48399faa69d5c60248771c98b059ac0473f40a7f
              state:
                  ----------
                  new:
                      running
                  old:
                      None

but then so-status will show the 'missing' changed to 'error'

`sudo so-status

Checking Docker status

    Docker ----------------------------------------------------------------------------------------------------------------------------------------- [ OK ]

Checking container statuses

    so-aptcacherng ---------------------------------------------------------------------------------------------------------------------------- [ PENDING ]
    so-cortex -------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-dockerregistry ------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-elastalert ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-elasticsearch ------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-filebeat ------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-fleet --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-grafana ------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-idstools ------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-influxdb ------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-kibana -------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-kratos -------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-logstash ------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-mysql --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-nginx --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-playbook --------------------------------------------------------------------------------------------------------------------------------- [ ERROR ]
    so-redis --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-sensoroni ----------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-soc ----------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-soctopus ------------------------------------------------------------------------------------------------------------------------------- [ MISSING ]
    so-telegraf ------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-thehive ------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-thehive-es ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-wazuh --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
`

I have also run salt-call state.highstate and this completes without error. 

I am stuck.

**Comment by defensivedepth** [comment, marked as answer]
Try running `sudo so-docker-refresh` & then `sudo salt-call state.highstate`

**Reply by bretthanus** [reply]
Thanks- ran both, then status again, and im back to the same aptcacherng pending and playbook/soctopus missing

**Reply by 8inaryMata1eao** [reply]
Fixed my issue.  Thanks Josh

**Reply by defensivedepth** [reply]
@bretthanus Can you please start a new Discussion? Also make sure to report if there are any errors in a highstate.
